BitList.cs

```diff
@@ -0,0 +1,1496 @@
+    /// </summary>
+    public class FileWritingException: FileException { }
+
+    /// <summary>
+    /// Exception generated when a BitList instance fails the conversion to another type.
+    /// </summary>
+    public class ConversionErrorException: Exception { }
+
+    /// <summary>
+    /// A list of bits.
+    /// </summary>
+    /// <exception cref="EmptyInstanceException" />
+    /// <exception cref="BooleanAlgebraException" />
+    /// <exception cref="FileReadingException" />
+    /// <exception cref="FileWritingException" />
+    /// <exception cref="ConversionErrorException" />
+    public sealed class BitList:
+        IList<bool>,
+        IEquatable<BitList>
+    {
+        private BitArray bits;
+
+        /// <summary>
+        /// No params ctor.
+        /// </summary>
+        public BitList()
+        {
+            bits = null;
+        }
+
+        /// <summary>
+        /// Ctor.
+        /// </summary>
+        /// <param name="bit">A bit.</param>
+        public BitList(bool bit)
+        {
+            bits = new BitArray(new[] { bit });
+        }
+
+        /// <summary>
+        /// Ctor.
+        /// </summary>
+        /// <param name="bits">A bit array.</param>
+        public BitList(bool[] bits)
+        {
+            this.bits = new BitArray(bits);
+        }
+
+        /// <summary>
+        /// Ctor.
+        /// </summary>
+        /// <param name="bits">A byte.</param>
+        public BitList(byte bits)
+        {
+            this.bits = new BitArray(new[] { bits });
+        }
+
+        /// <summary>
+        /// Ctor.
+        /// </summary>
+        /// <param name="bits">A byte array.</param>
+        public BitList(byte[] bits)
+        {
+            this.bits = new BitArray(bits);
+        }
+
+        /// <summary>
+        /// Ctor.
+        /// </summary>
+        /// <param name="bits">A 16 bit integer.</param>
+        public BitList(short bits)
+        {
+            this.bits = new BitArray(BitConverter.GetBytes(bits));
+        }
+
+        /// <summary>
+        /// Ctor.
+        /// </summary>
+        /// <param name="bits">A 16 bit integer array.</param>
+        public BitList(short[] bits)
+        {
+            //Convert the Int16 array to a Byte array and then create a new BitArray instance.
+            var bytesList = new List<byte[]>();
+            foreach (var num in bits)
+                bytesList.Add(BitConverter.GetBytes(num));
+            var byteList = new List<byte>();
+            foreach (var bytes in bytesList)
+            {
+                foreach (var num in bytes)
+                    byteList.Add(num);
+            }
+            this.bits = new BitArray(byteList.ToArray());
+        }
+
+        /// <summary>
+        /// Ctor.
+        /// </summary>
+        /// <param name="bits">A 32 bit integer.</param>
+        public BitList(int bits)
+        {
+            this.bits = new BitArray(new int[] { bits });
```

Comment:
Codacy found an issue: [Remove the array type; it is redundant.](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=5559704)